### PR TITLE
Customer post view

### PR DESCRIPTION
**Pull request by jens3cakes**

### CODE REVIEW

File: server/routes/posts.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 const express = require('express');
 const router = express.Router();
 const Post = require('../db/Models/Post');
-const Category = require('../db/Models/Category');
-const PostStatus = require('../db/Models/PostStatus');
-const PostPriority = require('../db/Models/PostPriority');
+const Category = require('../db/Models/Category')
```
Comment (taesup) [inline server/routes/posts.js]: These three lines are missing semi-colons

File: src/app/component/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -23,4 +31,15 @@ customer: any;
 
     ngOnInit() { }
 
+    getPosts() {
+        let postId = this.session.getCustomer()
```
Comment (taesup) [inline src/app/component/dashboard/dashboard.component.ts]: This variable name is misleading.

File: src/app/component/dashboard/dashboard.component.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
     <div class="links" *ngIf="isLoggedIn()">
         <div><a routerLink="/home"><img class="icon" src="https://image.flaticon.com/icons/svg/25/25694.svg"></a></div>
         <div><a routerLink="/messages"><img class="icon" src="https://cdn0.iconfinder.com/data/icons/thin-communication-messaging/24/thin-0306_chat_message_discussion_bubble_conversation-512.png"></a></div>
-        <div><a routerLink="/posts"><img class="icon" src="https://cdn3.iconfinder.com/data/icons/social-media-2172/128/posts-512.png"></a></div>
+        <div><a (click)="getPosts()" routerLink="/posts"><img class="icon" src="https://cdn3.iconfinder.com/data/icons/social-media-2172/128/posts-512.png"></a></div>
```
Comment (taesup) [inline src/app/component/dashboard/dashboard.component.html]: You should not use a click handler on a link. The link should just go to where it's pointed to. From that new page, you should handle how you need to pull in data for that page.

File: src/app/pages/home/home.component.ts [hunk 4]
```diff
@@ -9,44 +9,48 @@ import { Router } from '@angular/router';
...
-        city:'',
-        state:'',
-        zip_code:'',
-        email:'',
-        customer_id:null,
-        customerId:null,
-        can_bid:false,
-        budget:null
-      }
+    post:object;
```
Comment (taesup) [inline src/app/pages/home/home.component.ts]: Please define your post object using typescript

File: src/app/pages/postDetail/postDetail.component.ts [hunk 5]
```diff
@@ -8,7 +8,8 @@ import { Router, ActivatedRoute, ParamMap } from '@angular/router';
 })
 
 export class PostDetailComponent implements OnInit {
-  post:{
+  post:any
```
Comment (taesup) [inline src/app/pages/postDetail/postDetail.component.ts]: Please do not use the `any` type.

File: src/app/pages/posts/posts.component.ts [hunk 6]
```diff
@@ -9,6 +9,7 @@ import { SessionService } from '../../services/session.service'
 })
 
 export class PostsComponent implements OnInit {
+    posts:any;
```
Comment (taesup) [inline src/app/pages/posts/posts.component.ts]: Do not use the `any` type. Fully describe your objects using typescript

File: src/app/pages/posts/posts.component.ts [hunk 7]
```diff
@@ -44,11 +45,29 @@ post:{
...
 
-    ngOnInit() { }
+    ngOnInit() { 
+        let postId = this.route.snapshot.paramMap.get('id');
+    this.backend.getPostByCustomer(postId)
+      .then((resp) => {
+          this.posts = resp;
+        return this.posts
+      })
+    //   .catch((err) => {
```
Comment (taesup) [inline src/app/pages/posts/posts.component.ts]: If you don't need a catch here, you should remove this comment.

File: src/app/pages/posts/posts.component.ts [hunk 8]
```diff
@@ -44,11 +45,29 @@ post:{
...
+    //     return this.router.navigate(['/home'])
+    //   })
+    }
 
-    
+    fetchPostDetail(id){
+        //let postId = this.route.snapshot.paramMap.get('id');
```
Comment (taesup) [inline src/app/pages/posts/posts.component.ts]: Also remove this comment if you don't need it.

File: src/app/services/backend.service.ts [hunk 9]
```diff
@@ -35,8 +107,11 @@ export class BackendService {
   ) { }
 
   fetchPost(param) {
-    const searchUrl = this.baseUrl + `api/posts/${param}/edit`
+    const searchUrl = this.baseUrl + `api/posts/`+param;
     return this.http.get(searchUrl).toPromise()
+    .then((resp)=>{
```
Comment (taesup) [inline src/app/services/backend.service.ts]: This `.then` is essentially doing nothing. You should remove it.

File: src/app/services/post.service.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn:'root'
+})
+export class PostService{
+  baseUrl: string = "http://localhost:4200/";
+  post:any;
```
Comment (taesup) [inline src/app/services/post.service.ts]: You should describe your post using typescript.

**Review by taesup — COMMENTED** [review]
Left some feedback.

File: server/routes/posts.js (hunk 1 above)
Comment (taesup) [reply server/routes/posts.js]: Still missing semi colons here.

File: server/routes/vendors.js [hunk 11]
```diff
@@ -3,6 +3,7 @@ const router = express.Router();
 const validator = require('validator');
 const Vendor = require('../db/Models/Vendor');
 const bcrypt = require('bcryptjs');
+const Post = require('../db/Models/Post')
```
Comment (taesup) [inline server/routes/vendors.js]: Missing a semi colon

File: server/server.js [hunk 12]
```diff
@@ -109,10 +109,10 @@ passport.use('vendorLogin', new LocalStrategy((username, password, done) => {
 }));
 
 app.post('/api/customer/login', passport.authenticate('customerLocal', { failureRedirect: '' }),
-  function(req, res){
-     return res.send(req.user)
+  function (req, res) {
+    return res.send(req.user)
```
Comment (taesup) [inline server/server.js]: Missing semicolon.

File: src/app/component/dashboard/dashboard.component.html (hunk 3 above)
Comment (taesup) [reply src/app/component/dashboard/dashboard.component.html]: No click handlers on links. You should be doing the work that's in that click handler on the `/posts` page instead.

File: src/app/component/dashboard/dashboard.component.ts [hunk 13]
```diff
@@ -1,17 +1,25 @@
...
+
 @Component({
     selector: 'app-dashboard',
     templateUrl: './dashboard.component.html',
     styleUrls: ['./dashboard.component.scss']
 })
 
 export class DashboardComponent implements OnInit {
-customer: any;
+    customer: any;
```
Comment (taesup) [inline src/app/component/dashboard/dashboard.component.ts]: No using any.

File: src/app/component/dashboard/dashboard.component.ts [hunk 14]
```diff
@@ -1,17 +1,25 @@
...
+    posts: any
```
Comment (taesup) [inline src/app/component/dashboard/dashboard.component.ts]: No using any.

File: src/app/component/dashboard/dashboard.component.ts (hunk 2 above)
Comment (taesup) [reply src/app/component/dashboard/dashboard.component.ts]: This variable name is still misleading.

File: src/app/component/dashboard/dashboard.component.ts [hunk 15]
```diff
@@ -23,4 +31,15 @@ customer: any;
 
     ngOnInit() { }
 
+    getPosts() {
+        let postId = this.session.getCustomer()
+        return this.backend.getPostByCustomer(postId.id)
+            .then((resp) => {
+                this.posts = resp
```
Comment (taesup) [inline src/app/component/dashboard/dashboard.component.ts:65]: Missing semicolon.

File: src/app/pages/categoryPostPage/categoryPostPage.component.html [hunk 16]
```diff
@@ -0,0 +1,29 @@
+
+<div name="category" *ngFor="let post of posts">
+<h2>{{post.categoryId.name}}</h2>
+
+<h3>{{post.title}}</h3>
+<br>
```
Comment (taesup) [inline src/app/pages/categoryPostPage/categoryPostPage.component.html]: Don't use `<br>`. Instead wrap them in proper divs and use css to define the space between the text.

File: src/app/pages/categoryPostPage/categoryPostPage.component.ts [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+import { Router, ActivatedRoute, ParamMap } from '@angular/router';
+import { PostService } from '../../services/post.service'
+
+@Component({
+  templateUrl: './categoryPostPage.component.html',
+  styleUrls: ['./categoryPostPage.component.scss']
+})
+
+export class CategoryPostPage implements OnInit {
+posts:any;
```
Comment (taesup) [inline src/app/pages/categoryPostPage/categoryPostPage.component.ts]: No using any.

File: src/app/pages/home/home.component.ts (hunk 4 above)
Comment (Slagrimas) [reply src/app/pages/home/home.component.ts]: Jenn is working on that at the moment

File: src/app/pages/posts/posts.component.ts (hunk 7 above)
Comment (taesup) [reply src/app/pages/posts/posts.component.ts]: Please remove this comment if you don't need this code.

File: src/app/pages/postDetail/postDetail.component.ts (hunk 5 above)
Comment (Slagrimas) [reply src/app/pages/postDetail/postDetail.component.ts]: Jenn is working on that at the moment

File: src/app/pages/posts/posts.component.ts (hunk 8 above)
Comment (taesup) [reply src/app/pages/posts/posts.component.ts]: Again remove this comment if you're not using this piece of code.

File: src/app/services/backend.service.ts (hunk 9 above)
Comment (taesup) [reply src/app/services/backend.service.ts]: Remove this .then

File: src/app/pages/categoryPostPage/categoryPostPage.component.html (hunk 16 above)
Comment (jens3cakes) [reply src/app/pages/categoryPostPage/categoryPostPage.component.html]: The other Jen is going to address this when she does the css so she can manipulate it the way she wants.

**Pull request merged**